haskell/toolchain.bzl

```diff
@@ -72,7 +86,7 @@ def _run_ghc(hs, cc, inputs, outputs, mnemonic, arguments, params_file = None, e
         mnemonic = mnemonic,
         progress_message = progress_message,
         env = env,
-        arguments = [args] + arguments,
+        arguments = [],
```

Comment:
What's the reasoning for using `run_shell` instead of `run` with a parameters file like e.g. [here](https://github.com/bazelbuild/rules_dotnet/pull/47/files#diff-72490c7e47e72e5c8662d0f6408b43e1R204)?

Reply:
This is not possible here. `args.use_params_file(…)` instructs Bazel to dump all arguments into an auto-generated file, and replace them with a formatted string containing the path of the file. It works because `csc` can read arguments from so-called "response files", which are specified as file paths prefixed by the `@` character. It would be a good solution if GHC had a `--use-params-from-file=…` command-line parameter, but such is, to my knowledge, not the case.

Reply:
Thanks for explaining!